sympy/physics/mechanics/kane.py

```diff
@@ -118,7 +118,9 @@ class KanesMethod(_Methods):
     def __init__(self, frame, q_ind, u_ind, kd_eqs=None, q_dependent=None,
             configuration_constraints=None, u_dependent=None,
             velocity_constraints=None, acceleration_constraints=None,
-            u_auxiliary=None, bodies=None, forcelist=None):
+            u_auxiliary=None, bodies=None, forcelist=None,
+            explicit_kinematics=True
```

Comment:
You'll need to add an explanation of this and the new attributes to the docstring of KanesMethod.

Reply:
will do - but hopefully it can be made moot if we agree that the =False behavior should be the default behavior?

Reply:
Do we really need this kwarg? You can provide these matrices without adding a kwarg here. Right?

Reply:
I think we do need it: the behavior is different if one chooses to not solve for kinematics explicitly, and the method is limited to forces/velocities/accelerations only containing generalized speeds, not rates of generalized coordinates.